src/ScriptEditor/ui/ScriptEditorRoot.tsx

```diff
@@ -40,6 +40,18 @@ interface IProps {
 const openScripts: OpenScript[] = [];
 let currentScript: OpenScript | null = null;
 
+export function closeDeletedScripts(onlyHome: boolean): void {
+  for (let i = openScripts.length - 1; i >= 0; i--) {
+    const {hostname} = openScripts[i];
+    if (GetServer(hostname) === null || (onlyHome && hostname !== "home")) {
+      openScripts.splice(i, 1);
+    }
+  }
```

Comment:
Optional: It always bugs me a little to write a N^2 delete algorithm when the O(N) algorithm is basically as simple, even in places where the performance doesn't matter.
```suggestion
  let replaceIdx = 0;
  for (const script of openScripts) {
    openScripts[replaceIdx] = script;
    if (GetServer(script.hostname) !== null && (allServers || script.hostname === "home")) {
      replaceIdx++;
    }
  }
  openScripts.length = replaceIdx;
```

Reply:
Done